cassandra/util.py

```diff
@@ -1801,3 +1801,15 @@ def __gt__(self, other):
                 (is_major_ge and is_minor_ge and is_patch_ge and is_build_gt) or
                 (is_major_ge and is_minor_ge and is_patch_ge and is_build_ge and is_prerelease_gt)
                 )
+
+
+_query_timeout_regexp = re.compile('USING[ \t]+TIMEOUT[ \t]+[0-9]+[a-z]+[ \t]*;?$')
+
```

Comment:
We control all the places where the `add_timeout_to_query` function is called, there should be no need to have such a regex and performs checks with it.

If this is supposed to be a more universal function (and thus requiring this regex) then there are some problems with it (some of them unsolvable):

- For INSERT there may be no literal `USING TIMEOUT`. `INSERT INTO ks.t2 (pk, ck) VALUES (1, 2) USING TIMESTAMP 12345678 AND TIMEOUT 360ms;` is a valid statement. I don't think it's fixable using regular expressions.
- As soon as some new element of SELECT is added (and USING TIMEOUT doesn't have to be at the end of the statement) similar problem as above happens for SELECTs.
- Time unit can use uppercase letters, so `[a-z]` is wrong.
- `USING TIMEOUT` is case insensitive.

Reply:
agree, in worst case we will see that query fails on the integration and unit tests, regexp removed.